### ISSUE TITLE
Build Fail after adding and removing crosswalk, different versions

### ISSUE THREAD

**Issue by apavillet** [driftyco/ionic-cli#182]
Here's what I did

1. Adding crosswalk with ionic browser add crosswalk ( I suppose crosswalk 8 was added )
2. Removing it : ionic browser remove crosswalk
3. Re adding : ionic browser add crosswalk@version ( whatever this is it does not work from now on)

After the 2 and 3 => ionic run android produce the following

Just a thought to help : do you remove the ant-build and ant-gen folder when upgrading the version?

<code>
UNEXPECTED TOP-LEVEL EXCEPTION:
com.android.dex.DexException: Multiple dex files define Lcom/googlecode/eyesfree/braille/display/BrailleDisplayProperties$1;
	at com.android.dx.merge.DexMerger.readSortableTypes(DexMerger.java:596)
	at com.android.dx.merge.DexMerger.getSortedTypes(DexMerger.java:554)
	at com.android.dx.merge.DexMerger.mergeClassDefs(DexMerger.java:535)
	at com.android.dx.merge.DexMerger.mergeDexes(DexMerger.java:171)
	at com.android.dx.merge.DexMerger.merge(DexMerger.java:189)
	at com.android.dx.command.dexer.Main.mergeLibraryDexBuffers(Main.java:454)
	at com.android.dx.command.dexer.Main.runMonoDex(Main.java:303)
	at com.android.dx.command.dexer.Main.run(Main.java:246)
	at com.android.dx.command.dexer.Main.main(Main.java:215)
	at com.android.dx.command.Main.main(Main.java:106)

 FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':dexDebug'.
> com.android.ide.common.internal.LoggedErrorException: Failed to run command:
  	/home/homengo/Development/adt-bundle-linux-x86_64-20131030/sdk/build-tools/21.1.2/dx --dex --no-optimize --output /home/homengo/crosswalk-test/platforms/android/build/intermediates/dex/debug --input-list=/home/homengo/crosswalk-test/platforms/android/build/intermediates/tmp/dex/debug/inputList.txt
  Error Code:
  	2
  Output:
  	
  	UNEXPECTED TOP-LEVEL EXCEPTION:
  	com.android.dex.DexException: Multiple dex files define Lcom/googlecode/eyesfree/braille/display/BrailleDisplayProperties$1;
  		at com.android.dx.merge.DexMerger.readSortableTypes(DexMerger.java:596)
  		at com.android.dx.merge.DexMerger.getSortedTypes(DexMerger.java:554)
  		at com.android.dx.merge.DexMerger.mergeClassDefs(DexMerger.java:535)
  		at com.android.dx.merge.DexMerger.mergeDexes(DexMerger.java:171)
  		at com.android.dx.merge.DexMerger.merge(DexMerger.java:189)
  		at com.android.dx.command.dexer.Main.mergeLibraryDexBuffers(Main.java:454)
  		at com.android.dx.command.dexer.Main.runMonoDex(Main.java:303)
  		at com.android.dx.command.dexer.Main.run(Main.java:246)
  		at com.android.dx.command.dexer.Main.main(Main.java:215)
  		at com.android.dx.command.Main.main(Main.java:106)
  	

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 2 mins 18.443 secs

/home/homengo/crosswalk-test/platforms/android/cordova/node_modules/q/q.js:126
                    throw e;
                          ^
Error code 1 for command: /home/homengo/crosswalk-test/platforms/android/gradlew with args: assembleDebug,-b,/home/homengo/crosswalk-test/platforms/android/build.gradle,-Dorg.gradle.daemon=true
Error: /home/homengo/crosswalk-test/platforms/android/cordova/run: Command failed with exit code 8
    at ChildProcess.whenDone (/usr/lib/node_modules/cordova/node_modules/cordova-lib/src/cordova/superspawn.js:135:23)
    at ChildProcess.emit (events.js:98:17)
    at maybeClose (child_process.js:756:16)
    at Process.ChildProcess._handle.onexit (child_process.js:823:5)

</code>

**Comment by AlexYankee** [comment]
Same here

**Comment by furkanayhan** [comment]
same problem

**Comment by colllin** [comment]
Try reloading your terminal.  If that doesn't help, try upgrading the moving parts listed by `ionic info`.

**Comment by colllin** [comment]
BUILD SUCCESSFUL!  It's a miracle.  I think it's because I had two versions of crosswalk downloaded. So I removed all of the crosswalk stuff and started over.

Steps I took:

```
$ ionic browser remove crosswalk
$ rm -rf engine
$ rm -rf platforms/android                       # careful with this one if you made local changes.
$ ionic platform android
$ ionic browser add crosswalk@10.39.235.15
$ ionic run android

**Comment by fabiosussetto** [comment]
@colllin thanks for that, it fixed the issue for us.

**Comment by fabiosussetto** [comment]
Why has this been closed? The solution proposed is more like a workaround, isn't it?

**Comment by jbavari** [comment]
@fabiosussetto - sorry, didn't mean to close with any bad intent.  Just wasn't sure how/what to fix here.

I'd love some help figuring out, as I'm not really able to reproduce it.

Are you able to reproduce it following the steps the original poster did? Did the adding the version mess it up?

**Comment by fabiosussetto** [comment]
@jbavari: no worries mate, I was wondering if it's been fixed or what. Tomorrow I'll see if I can find a good way for you to reproduce this.

**Comment by colllin** [comment]
I think the problem is because people are reading down the blog post and
typing in everything they see. At least that's what I did. First I added
crosswalk via the cli. Then a second later I was informed that I could list
all available versions, so I did. Whoa, it installed 37.x but there's a
whole 2 major versions newer, 39.x!!!  I'll add that one now via the cli,
just like the next example in the blog post. (Note that I haven't removed
the original version first, or attempted to build anything).

And that might explain why removing the engines folder fixes it. Then I can
start over and only install the 39.x version, so nothing gets confused.
